app/controllers/shopping_cart/carts_controller.rb

```diff
@@ -0,0 +1,21 @@
+require_dependency "shopping_cart/application_controller"
```

Comment:
better to move all requirements into lib/shopping_cart.rb or lib/shopping_cart/engine.rb